enhancements/sig-architecture/224-event-based-manifestwork/README.md

```diff
@@ -0,0 +1,788 @@
+    // <resource-type> refers to the resource included in the cloud event payload.
+    "source": "/sources/<source-id>/clusters/<cluster-id>/<resource-type>",
+    // Use "<reverse-domain-name-of-resource-organization>.<resource-version>.<resource-name>.<subsource>.<action>" as the format for this attribute, which contains a value describing the type of event related to the originating occurrence.
+    // For example, "io.open-cluster-management.v1.manifestwork.statusresync.requested" can be used as a sample format.
+    "type": "<reverse-domain-name-of-resource-organization>.<resource-version>.<resource-name>.<subsource>.<action>",
+    // Timestamp of when the occurrence happened. Optional, if present, MUST adhere to the format specified in RFC 3339.
+    "time": "<timestamp-of-the-event-occurrence>",
+    // Content type of data value, If present, MUST adhere to the format specified in RFC 2046.
+    // By default "application/json" will be used if it is not set.
+    "datacontenttype": "<data-content-type>"
+}
+// Data
+{
+    "statusHashes": [
+        {
+        "resourceID": "748d04ad-887a-43aa-b123-da1373227011",
+        "statusHash": "12345abc"
+        },
+        {
+        "resourceID":  "858d03ad-997b-52bc-b456-ab1473228022",
+        "statusHash": "abc12345"
+        }
+    ]
+}
+```
+
+### Failure Handling
+
+~~- Retry Mechanism(TBD):~~
+
+~~In the event of a failure during event delivery or processing, a retry mechanism should be implemented to ensure reliable event propagation. When an event fails to reach its destination or encounters an error during processing, it should be retried for a certain number of times before giving up.~~
+
+~~The retry mechanism should employ exponential backoff, gradually increasing the delay between retries to prevent overwhelming the system with failed events. Each retry attempt should be logged, including relevant information such as the event payload, timestamp, and error details.~~
+
+- Resync Mechanism:
+
+To ensure the integrity of the event-based manifestworks system during source or work agent restart or reconnection, the following optimized resync mechanism is employed:
+
+1. On start, the source subscribes to the spec resync event topic /sources/resync/+/manifests from the broker.
+2. On start, the work agent subscribes to the status resync event topic /sources/+/resync/clusters/manifestsstatus from the broker.
+3. Upon work agent restart or reconnect, the work agent generates a spec resync event with resource versions and sends it to the broker.
+4. Upon source restart or reconnect, the source calculates a status hash of its owned resources and sends a status resync event with the resource hash to the broker.
+5. The source compares the resource versions in the spec resync event message with the hub's state and generates spec events to the broker for the work agent to receive missing spec events.
```

Comment:
As said above, we need to be clear on where/how is hub storing this state. This should not be Kube API

Reply:
For the manifestwork itself, we store it in cache. For high level API that abstract manifestwork API, they will be saved in DB or ETCD, depending on different kinds of Source arahitecture, but that's out of the scope of the proposal.

Reply:
Agreed. High level API in etcd. The manifestwork and state in cache can be a  little problematic IMO. But let it go for now as its outside scope.